invertedindex/core_index.go

```diff
@@ -0,0 +1,74 @@
+package invertedindex
```

Comment:
Let's avoid package sprawl. Delete this package and move the types to the existing `index` package, with the proposed renames?